### ISSUE TITLE
All classification metrics are the same when finetuning network with flash

### ISSUE THREAD

**Issue by mattwarkentin** [PyTorchLightning/metrics#543]
## 🐛 Bug

Hi,

I am trying to play around with the `ImageClassifier` finetune example from `flash`. However, when I finetune the network, all of my classification metrics are the exact same for any given epoch. Not sure whether this is a bug in my code or something about how `flash`, `torchmetrics`, and the `bolts` callback work. The code below should reproduce based on a slightly adapted version of the example shown here: https://lightning-flash.readthedocs.io/en/latest/reference/image_classification.html#example.

### To Reproduce

```py
from flash.image import ImageClassifier, ImageClassificationData
from flash.core.data.utils import download_data
from pl_bolts.callbacks import PrintTableMetricsCallback
import flash
import torchmetrics as tm

download_data("https://pl-flash-data.s3.amazonaws.com/hymenoptera_data.zip", "./data")

datamodule = ImageClassificationData.from_folders(
    train_folder="data/hymenoptera_data/train/",
    val_folder="data/hymenoptera_data/val/",
    batch_size=16
)

# Threshold at zero since logits are returned
mets = [
  tm.Accuracy(num_classes=datamodule.num_classes, threshold=0), 
  tm.Recall(num_classes=datamodule.num_classes, threshold=0), 
  tm.Precision(num_classes=datamodule.num_classes, threshold=0), 
  tm.F1(num_classes=datamodule.num_classes, threshold=0)
]

model = ImageClassifier(
  backbone='resnet18', 
  num_classes=datamodule.num_classes, 
  pretrained=False, 
  metrics=mets
)

ptb = PrintTableMetricsCallback()

trainer = flash.Trainer(max_epochs=2, callbacks=ptb)

trainer.finetune(model, datamodule=datamodule, strategy="freeze")
```
This is printed after the second epoch. It can be seen that the value for all 4 metrics are the same within an epoch, and also the callback prints the metrics in duplicate rows for some reason.
```
val_accuracy│val_recall│val_precision│val_f1│val_cross_entropy                                                                             
──────────────────────────────────────────────────────────────
0.0│0.0│0.0│0.0│1.4804610013961792
0.4575163424015045│0.4575163424015045│0.4575163424015045│0.4575163424015045│0.735869824886322
0.4575163424015045│0.4575163424015045│0.4575163424015045│0.4575163424015045│0.735869824886322
0.6470588445663452│0.6470588445663452│0.6470588445663452│0.6470588445663452│0.6557145714759827
0.6470588445663452│0.6470588445663452│0.6470588445663452│0.6470588445663452│0.6557145714759827
```

### Expected behavior

I expect the metrics to be different.

### Environment

- PyTorch Version: `1.8.1`
- PyTorch Lightning Version: `1.4.7`
- Lightning Flash Version: `0.5.0`
- Lightning Bolts Version: `0.4.0`
- Torchmetrics Version: `0.5.0`
- OS: MacOS
- Python version: `Python 3.7.0`

**Comment by SkafteNicki** [comment]
Try changing the `average` method from `micro` to `macro`. Under the default `micro` average the formula for the metrics may be the same.

**Comment by dreaquil** [comment]
I came across this same problem. Your solution didn't work and I haven't really understood why micro would be an issue. Would you mind explaining further please? specificity, recall and accuracy are showing identical metrics. F1 is now different after the recommended change.

![image](https://user-images.githubusercontent.com/32460579/139898989-71457a00-764d-4bc9-a88f-f50035b35021.png)

**Comment by Waterkin** [comment]
have the same value for `accuracy`
